hil/src/flash.rs

```diff
@@ -65,16 +72,56 @@ fn extract(path_to_rts: &Utf8Path) -> Result<TempDir> {
     Ok(temp_dir)
 }
 
-fn flash_cmd(variant: FlashVariant, extracted_dir: &Path) -> Result<()> {
+fn generate_random_files<R: rand::Rng>(output_dir: &Path, rng: &mut R) -> Result<()> {
```

Comment:
nit: this is fine but `rng: &mut impl rand::Rng)` is arguably easier. functions ideally have as few generics as possible, so impl can help with that sometimes.